src/services/structures/structures.class.js

```diff
@@ -325,14 +325,17 @@ exports.Structures = class Structures extends Service {
           WHERE id = $1`,
           [id, email]);
           await db.collection('structures').updateOne({ _id: new ObjectID(structureId) }, { $set: { 'contact.email': email } });
-          await db.collection('users').updateOne({ 'name': structure.contact.email, 'entity.$id': new ObjectID(structureId) }, { $set: { name: email } });
+          await db.collection('users').updateOne(
+            { 'name': structure.contact.email, 'entity.$id': new ObjectID(structureId), 'roles': 'structure' },
```

Comment:
@Ornella452 humm plutot `'roles': { $in: ['structure'] }` comme c'est un array

Reply:
bien vu !